uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/IdentityZoneEndpointsMockMvcTests.java

```diff
@@ -1889,6 +1897,190 @@ public void userCanReadAZone_withZoneZoneIdReadToken() throws Exception {
         assertEquals("kid", zoneResult.getConfig().getTokenPolicy().getActiveKeyId());
     }
 
+    @Test
+    public void testCreateKeyProviderConfigDeniedForSystemZone() throws Exception {
+        KeyProviderConfig keyProviderConfig = new KeyProviderConfig("clientId", "dcsTenantId");
+        getMockMvc().perform(
+                post("/identity-zones/" + IdentityZoneHolder.getUaaZone().getId() + "/key-provider-config")
+                        .header("Authorization", "Bearer " + adminToken)
+                        .contentType(APPLICATION_JSON)
+                        .accept(APPLICATION_JSON)
+                        .content(JsonUtils.writeValueAsString(keyProviderConfig)))
+                .andExpect(status().isForbidden())
+                .andReturn().getResponse();
+    }
+
+    @Test
+    public void testGetKeyProviderConfigDeniedForSystemZone() throws Exception {
+        getMockMvc().perform(
+                get("/identity-zones/" + IdentityZoneHolder.getUaaZone().getId() + "/key-provider-config/1")
+                        .header("Authorization", "Bearer " + adminToken)
+                        .contentType(APPLICATION_JSON)
+                        .accept(APPLICATION_JSON))
+                .andExpect(status().isForbidden())
+                .andReturn().getResponse();
+    }
+
+    @Test
+    public void testFindKeyProviderConfigDeniedForSystemZone() throws Exception {
```

Comment:
Should GET keyproviderconfig be forbidden for system zone? We can GET base zone.

Reply:
More fundamentally, since we cannot change keyproviderconfig through API, we either have to enable provisioning through uaa yml, or give up base zone JwtBearer zonified flow entirely

Reply:
This applies to `CREATE` as well. Does it make sense to add/retrieve KeyProviderConfig for base zone? We didn't think it is a valid use case. The fallback approach in `JwtBearer` flow will be to use global config ---which doesn't fit the same model as the zone `ZoneKeyProviderConfig`

Reply:
yes, add/retrieve KeyProviderConfig for base zone does not make sense